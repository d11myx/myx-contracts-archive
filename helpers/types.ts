import { IPool } from '../types';
import { eNetwork } from './constants';
import { IFundingRate } from '../types/contracts/core/FundingRate';
import type { PromiseOrValue } from '../types/common';
import type { BigNumberish } from 'ethers';

export interface SymbolMap<T> {
    [symbol: string]: T;
}

export type ParamsPerNetwork<T> = {
    [k in eNetwork]?: T;
};

export type TradingFeeTier = {
    takerFee: PromiseOrValue<BigNumberish>;
    makerFee: PromiseOrValue<BigNumberish>;
};

export interface PairInfoConfig {
    pairTokenDecimals: number;
    useWrappedNativeToken: boolean;
    pair: IPool.PairStruct;
    tradingConfig: IPool.TradingConfigStruct;
    tradingFeeConfig: IPool.TradingFeeConfigStruct & TradingFeeTier;
    fundingFeeConfig: IFundingRate.FundingFeeConfigStruct;
}

export interface ReserveConfiguration {
    MarketTokenSymbol: string;
    MarketTokenName: string;
    MarketTokenDecimals: number;
    MarketTokenAddress: ParamsPerNetwork<string>;
    WrapperTokenAddress: ParamsPerNetwork<string>;
    PairsConfig: SymbolMap<PairInfoConfig>;
    PairAssets: ParamsPerNetwork<SymbolMap<string>>;
    ExecuteOrderTimeDelay: number;
    OraclePriceFeedAddress: ParamsPerNetwork<string>;
    OraclePriceId: ParamsPerNetwork<SymbolMap<string>>;
    UniswapRouterAddress: ParamsPerNetwork<string>;
    UniswapTokenPathConfig: ParamsPerNetwork<SymbolMap<string>>;
}
