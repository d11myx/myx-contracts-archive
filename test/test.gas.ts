import { ethers, waffle } from 'hardhat';
import { expect } from './shared/expect';

import snapshotGasCost from './shared/snapshotGasCost';
import { TestGas, TestMulticall } from '../types';

describe('TestGas', async () => {
    const wallets = waffle.provider.getWallets();

    let testGas: TestGas;

    beforeEach('create multicall', async () => {
        const testGasFacory = await ethers.getContractFactory('TestGas');
        testGas = (await testGasFacory.deploy()) as TestGas;
    });
    it('', async () => {
         expect(await testGas.owner()).to.be.eq(wallets[0].address);
    });

    it('gas cost test maping(address=>struct)', async () => {
        await snapshotGasCost(testGas.saveIncreasePosit());
    });

    it('gas cost of saveOrderWithTpSl', async () => {
        await snapshotGasCost(testGas.saveOrderWithTpSl());
    });

    it('gas cost of testKey', async () => {
        await snapshotGasCost(testGas.testKey('1'));
    });

    it('gas cost of testKeys', async () => {
        await snapshotGasCost(testGas.testKeys('1'));
    });
});
