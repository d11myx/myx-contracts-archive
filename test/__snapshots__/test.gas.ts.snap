// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestGas gas cost of calldata params 1`] = `24865`;

exports[`TestGas gas cost of infos 1`] = `50286`;

exports[`TestGas gas cost of memory params 1`] = `26464`;

exports[`TestGas gas cost of saveOrderWithTpSl 1`] = `112103`;

exports[`TestGas gas cost of saveUint32Tests 1`] = `43645`;

exports[`TestGas gas cost of saveUint256Tests 1`] = `43628`;

exports[`TestGas gas cost of testKey 1`] = `43575`;

exports[`TestGas gas cost of testKeys 1`] = `45871`;

exports[`TestGas gas cost test maping(address=>struct) 1`] = `178707`;
